lasagne/tests/layers/test_recurrent.py

```diff
@@ -94,15 +94,78 @@ def test_recurrent_init_val_error():
+        (n_batch, n_steps, n_channels, width, height))
+    l_in_to_hid = lasagne.layers.Conv2DLayer(
+        lasagne.layers.InputLayer((None, n_channels, width, height)),
+        n_out_filters, filter_shape, pad='same')
+    l_hid_to_hid = lasagne.layers.Conv2DLayer(
+        lasagne.layers.InputLayer((None, n_out_filters, width, height)),
+        n_out_filters, filter_shape, pad='same')
+    l_rec = lasagne.layers.CustomRecurrentLayer(
+        l_in, l_in_to_hid, l_hid_to_hid)
+    out = theano.function([l_in.input_var], lasagne.layers.get_output(l_rec))
+    out_shape = out(np.zeros((n_batch, n_steps, n_channels, width, height),
+                             dtype=theano.config.floatX)).shape
+    assert out_shape == (n_batch, n_steps, n_out_filters, width, height)
+
+
 def test_recurrent_init_shape_error():
-    # check if errors are raised if output shaped for subnetworks are not
-    # correct
-    num_hid = 5
+    # Check that the custom recurrent layer throws errors for invalid shapes
+    n_batch, n_steps, n_channels, width, height = (2, 3, 4, 5, 6)
+    n_out_filters = 7
+    filter_shape = (3, 3)
+    l_in = lasagne.layers.InputLayer(
+        (n_batch, n_steps, n_channels, width, height))
+    l_hid_to_hid = lasagne.layers.Conv2DLayer(
+        lasagne.layers.InputLayer((n_batch, n_out_filters, width, height)),
+        n_out_filters, filter_shape, pad='same')
+
+    # When precompute_input == True, input_to_hidden.shape[0] must be None
+    # or n_batch*n_steps
+    l_in_to_hid = lasagne.layers.Conv2DLayer(
+        lasagne.layers.InputLayer((n_batch, n_channels, width, height)),
+        n_out_filters, filter_shape, pad='same')
+    with pytest.raises(ValueError):
+        l_rec = lasagne.layers.CustomRecurrentLayer(
+            l_in, l_in_to_hid, l_hid_to_hid, precompute_input=True)
+
+    # When precompute_input = False, input_to_hidden.shape[1] must be None
+    # or hidden_to_hidden.shape[1]
+    l_in_to_hid = lasagne.layers.Conv2DLayer(
+        lasagne.layers.InputLayer((n_batch + 1, n_channels, width, height)),
+        n_out_filters, filter_shape, pad='same')
+    with pytest.raises(ValueError):
+        l_rec = lasagne.layers.CustomRecurrentLayer(
+            l_in, l_in_to_hid, l_hid_to_hid, precompute_input=False)
+
+    # In any case, input_to_hidden and hidden_to_hidden's output shapes after
+    # the first dimension must match
+    l_in_to_hid = lasagne.layers.Conv2DLayer(
+        lasagne.layers.InputLayer((None, n_channels, width + 1, height)),
+        n_out_filters, filter_shape, pad='same')
+    with pytest.raises(ValueError):
+        l_rec = lasagne.layers.CustomRecurrentLayer(
+            l_in, l_in_to_hid, l_hid_to_hid)
+
+    # And, the output shape of input_to_hidden must match the input shape
+    # of hidden_to_hidden past the first dimension.  By not using padding,
+    # the output of l_in_to_hid will be cropped, which will make the
+    # shape inappropriate.
```

Comment:
"the output of l_in_to_hid" -> "the output of l_hid_to_hid"

The former wouldn't be a problem at all, the latter is.